src/xenia/kernel/xam/xam_content.cc

```diff
@@ -31,6 +31,64 @@ namespace xe {
 namespace kernel {
 namespace xam {
 
+class XStaticContentEnumerator : public XEnumerator {
+ public:
+  XStaticContentEnumerator(KernelState* kernel_state,
+                           size_t items_per_enumerate)
+      : XEnumerator(kernel_state, items_per_enumerate, sizeof(XCONTENT_DATA)) {}
+
+  void AppendItem(ContentData item) { items_.push_back(item); }
+  size_t item_count() const { return items_.size(); }
+
+  uint32_t WriteItems(uint32_t buffer_ptr, uint8_t* buffer_data,
+                      uint32_t buffer_size, uint32_t* written_count) override {
+    size_t count =
+        std::min(items_.size() - current_item_, items_per_enumerate());
+
+    if (!count) {
+      return X_ERROR_NO_MORE_FILES;
+    }
+
+    size_t items_size = count * item_size();
+    size_t actual_buffer_size = buffer_size;
+
+    if (actual_buffer_size < item_size()) {
+      // Game provided max amount of items instead of used space
+      if (count > actual_buffer_size) {
+        return X_ERROR_INSUFFICIENT_BUFFER;
+      }
+      actual_buffer_size = items_size;
+    }
+
+    if (items_size > actual_buffer_size) {
+      return X_ERROR_INSUFFICIENT_BUFFER;
+    }
+    memset(buffer_data, 0, items_size);
+
+    auto details = reinterpret_cast<XCONTENT_DATA*>(buffer_data);
+
+    for (size_t i = 0, o = current_item_; i < count; ++i, ++current_item_) {
+      const auto& item = items_[current_item_];
+      details[i].content_type = item.content_type;
```

Comment:
The `ContentData` type already has a helper function for (correctly) writing out to guest memory. You don't need to do it yourself here.

```cpp
item.Write(&details[i]);
```